Can't escape [] in format line

#### Current Behavior
<!-- A clear and concise description of the behavior. -->
Escaping doesn't seem to work as explained.

#### Expected Behavior
<!-- A clear and concise description of what you expected to happen. -->
The `[` and `]` to be escaped.

#### Additional context/Screenshots
<!-- Add any other context about the problem here. If applicable, add screenshots to help explain. -->
> Note that toml has its own escape syntax. 

In the wiki this links to a README which is missing the section about escaping, etc.
https://starship.rs/config/#format-strings

#### Possible Solution
<!--- Only if you have suggestions on a fix for the bug -->

#### Environment
- Starship version: 1.0.0
- zsh version: zsh 5.8 (x86_64-pc-linux-gnu)
- Operating system: Arch Linux Rolling Release
- Terminal emulator: <unknown terminal> <unknown version>
- Git Commit Hash: dfbf861f
- Branch/Tag: v1.0.0
- Rust Version: rustc 1.56.1 (59eed8a2a 2021-11-01)
- Rust channel:  release
- Build Time: 2021-11-09 21:35:52 +00:00
#### Relevant Shell Configuration

```bash
# Path to your oh-my-zsh installation.
export ZSH="/home/xo/.oh-my-zsh"

# Set name of the theme to load --- if set to "random", it will
# load a random theme each time oh-my-zsh is loaded, in which case,
# to know which specific one was loaded, run: echo $RANDOM_THEME
# See https://github.com/ohmyzsh/ohmyzsh/wiki/Themes
ZSH_THEME="robbyrussell"

# Uncomment the following line to change how often to auto-update (in days).
zstyle ':omz:update' frequency 13

# Uncomment the following line if pasting URLs and other text is messed up.
DISABLE_MAGIC_FUNCTIONS="true"

# Which plugins would you like to load?
# Standard plugins can be found in $ZSH/plugins/
# Custom plugins may be added to $ZSH_CUSTOM/plugins/
# Example format: plugins=(rails git textmate ruby lighthouse)
# Add wisely, as too many plugins slow down shell startup.
plugins=(git zsh-autosuggestions zsh-syntax-highlighting)

source $ZSH/oh-my-zsh.sh

# You may need to manually set your language environment
export LANG=en_US.UTF-8

# Preferred editor
export EDITOR='nano'

# Preferred terminal
export TERMINAL=alacritty

# For a full list of active aliases, run `alias`.
source ~/.zsh/aliases

# Setup node
export NVM_DIR="$HOME/.nvm"
[ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"  # This loads nvm
[ -s "$NVM_DIR/bash_completion" ] && \. "$NVM_DIR/bash_completion"  # This loads nvm bash_completion

export STARSHIP_CONFIG=~/.config/starship.toml
eval "$(starship init zsh)"
```

#### Starship Configuration

```toml
# Inserts a blank line between shell prompts
add_newline = true

# Allow git to work in larger repos
command_timeout = 2000

# Replace the "❯" symbol in the prompt with "➜"
[character]                            # The name of the module we are configuring is "character"
success_symbol = "[➜](bold green)"     # The "success_symbol" segment is being set to "➜" with the color "bold green"
error_symbol = "[✗](bold red) "

[hostname]
ssh_only = false
format =  "[\[$hostname](bold red)\] "
trim_at = ".unraid.net"
disabled = false
```

I also tried `\[[$hostname](bold red)\] ` with the same error.

> [ERROR] - (starship::config): Unable to parse the config file: invalid escape character in string: `[` at line 14 column 14

--- a/docs/config/README.md
+++ b/docs/config/README.md
@@ -65,7 +65,7 @@ You can use texts, variables and text groups in a format string.
 #### Variable
 
 A variable contains a `$` symbol followed by the name of the variable.
-The name of a variable only contains letters, numbers and `_`.
+The name of a variable can only contain letters, numbers and `_`.
 
 For example:
 
@@ -112,21 +112,13 @@ For example:
 - When `$all` is a shortcut for `\[$a$b\] `, `($all)` will show nothing only if `$a` and `$b` are both `None`.
   This works the same as `(\[$a$b\] )`.
 
-#### Escapable characters
+#### Special characters
 
-The following symbols have special usage in a format string.
-If you want to print the following symbols, you have to escape them with a backslash (`\`).
+The following symbols have special usage in a format string and must be escaped: `$ \ [ ] ( )`.
 
-- \$
-- \\
-- [
-- ]
-- (
-- )
-
-Note that `toml` has [its own escape syntax](https://github.com/toml-lang/toml#user-content-string).
-It is recommended to use a literal string (`''`) in your config.
-If you want to use a basic string (`""`), pay attention to escape the backslash `\`.
+Note that TOML has [both basic strings and literal strings](https://toml.io/en/v1.0.0#string).
+It is recommended to use a literal string (surrounded by single quotes) in your config.
+If you want to use a basic string (surrounded by double quotes), you must escape the backslash itself (i.e. use `\\`).
 
 For example, when you want to print a `$` symbol on a new line, the following configs for `format` are equivalent:
 

